cmd/kube-proxy/app/server.go

```diff
@@ -310,73 +483,84 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
-	endpointsConfig.RegisterEventHandler(endpointsEventHandler)
-	go endpointsConfig.Run(wait.NeverStop)
-
-	// This has to start after the calls to NewServiceConfig and NewEndpointsConfig because those
-	// functions must configure their shared informer event handlers first.
-	go informerFactory.Start(wait.NeverStop)
-
-	config.NodeRef = &clientv1.ObjectReference{
+	nodeRef := &clientv1.ObjectReference{
 		Kind:      "Node",
 		Name:      hostname,
 		UID:       types.UID(hostname),
 		Namespace: "",
 	}
 
-	conntracker := realConntracker{}
-
-	return NewProxyServer(client, eventClient, config, iptInterface, proxier, eventBroadcaster, recorder, conntracker, proxyMode)
+	return &ProxyServer{
+		Client:                 client,
+		EventClient:            eventClient,
+		IptInterface:           iptInterface,
+		Proxier:                proxier,
+		Broadcaster:            eventBroadcaster,
+		Recorder:               recorder,
+		ConntrackConfiguration: o.config.Conntrack,
+		Conntracker:            &realConntracker{},
+		ProxyMode:              proxyMode,
+		NodeRef:                nodeRef,
+		HealthzBindAddress:     o.config.HealthzBindAddress,
+		OOMScoreAdj:            o.config.OOMScoreAdj,
+		ResourceContainer:      o.config.ResourceContainer,
+		ConfigSyncPeriod:       o.config.ConfigSyncPeriod.Duration,
+		ServiceEventHandler:    serviceEventHandler,
+		ServiceHandler:         serviceHandler,
+		EndpointsEventHandler:  endpointsEventHandler,
+	}, nil
 }
 
 // Run runs the specified ProxyServer.  This should never exit (unless CleanupAndExit is set).
 func (s *ProxyServer) Run() error {
 	// remove iptables rules and exit
-	if s.Config.CleanupAndExit {
+	if s.CleanupAndExit {
 		encounteredError := userspace.CleanupLeftovers(s.IptInterface)
 		encounteredError = iptables.CleanupLeftovers(s.IptInterface) || encounteredError
 		if encounteredError {
-			return errors.New("Encountered an error while tearing down rules.")
+			return errors.New("encountered an error while tearing down rules.")
 		}
 		return nil
 	}
 
+	// TODO(vmarmol): Use container config for this.
+	var oomAdjuster *oom.OOMAdjuster
+	if s.OOMScoreAdj != nil {
+		oomAdjuster = oom.NewOOMAdjuster()
+		if err := oomAdjuster.ApplyOOMScoreAdj(0, int(*s.OOMScoreAdj)); err != nil {
+			glog.V(2).Info(err)
+		}
+	}
+
+	if len(s.ResourceContainer) != 0 {
+		// Run in its own container.
+		if err := resourcecontainer.RunInResourceContainer(s.ResourceContainer); err != nil {
+			glog.Warningf("Failed to start in resource-only container %q: %v", s.ResourceContainer, err)
+		} else {
+			glog.V(2).Infof("Running in resource-only container %q", s.ResourceContainer)
+		}
+	}
+
 	s.Broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: s.EventClient.Events("")})
 
 	// Start up a webserver if requested
-	if s.Config.HealthzPort > 0 {
+	if len(s.HealthzBindAddress) > 0 {
```

Comment:
It seems that the deprecated healthzPort flag is being handled by applyDeprecatedHealthzPortToConfig(). So it would behave the same as before. This LGTM.